### PR TITLE
When the recommendation is clicked, navigate to its url

### LINKED ISSUE #40: handle clicks on the recommendation (recommendation view)
When the recommendation el is clicked, ask the urlbar to navigate to the current recommendation URL.

Seems straightforward.

### PR DESCRIPTION
**Pull request by jaredhirsch**
@chuckharmston R? This one depends on PR #91, seems to work reliably.

Fixes #40.

### CODE REVIEW

File: lib/ui/recommendation.js [hunk 1]
```diff
@@ -144,9 +144,20 @@ Recommendation.prototype = {
...
-    // the URL should already be set in the urlbar by the time the event is received.
+    // Only navigate if it's a left click.
+    if (evt.button !== 0) {
+      return;
+    }
+
+    // Unfortunately, because the urlbar state isn't set by mousing over an
+    // item, we have to set the urlbar state, then ask it to navigate.
+    // Set state by faking a selection event...
+    const url = this.el.querySelector('#universal-search-recommendation-url').textContent;
```
Comment (chuckharmston) [inline lib/ui/recommendation.js]: Yeah, let's use `data-` attributes. Feels weird reading `textContent` for meaningful data.

**Comment by chuckharmston** [comment]
Merge away!

File: lib/ui/recommendation.js (hunk 1 above)
Comment (jaredhirsch) [inline lib/ui/recommendation.js]: :+1: sounds good

Comment (jaredhirsch) [inline lib/ui/recommendation.js]: Funny, of course XUL doesn't support html5 `dataset`, so we have to use getAttribute/setAttribute. Still better than fishing out textContent.

**Pull request merged**